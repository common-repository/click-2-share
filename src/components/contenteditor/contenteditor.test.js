import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import ContentEditor from './contenteditor';

describe( 'Content Editor Component', () => {
	const mockSetAttributes = jest.fn();

	test( 'User can type into the box', async () => {
		const user = userEvent.setup();
		const oldText = 'Hello';
		const newText = ' World !';

		render(
			<ContentEditor
				post={ oldText }
				setAttributes={ mockSetAttributes }
			/>
		);

		const textbox = screen.getByRole( 'textbox' );

		expect( textbox ).toBeInTheDocument();
		expect( textbox ).toHaveTextContent( oldText );

		await user.type( textbox, newText );
		expect( textbox ).toHaveTextContent( oldText + newText );
	} );

	test( 'Matches the Snapshot', () => {
		const text = 'I want to share this on Social Media!';

		const { asFragment } = render(
			<ContentEditor post={ text } setAttributes={ mockSetAttributes } />
		);

		expect( asFragment() ).toMatchSnapshot();
	} );
} );
