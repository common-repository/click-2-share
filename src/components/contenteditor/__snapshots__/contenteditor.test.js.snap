// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Editor Component Matches the Snapshot 1`] = `
<DocumentFragment>
  <p
    aria-label="Write your post hereâ€¦"
    aria-multiline="true"
    aria-readonly="false"
    class="block-editor-rich-text__editable rich-text"
    contenteditable="true"
    role="textbox"
    style="white-space: pre-wrap; min-width: 1px;"
  >
    I want to share this on Social Media!
  </p>
</DocumentFragment>
`;
